packages/data/src/options/with-options-hydration.js

```diff
@@ -10,37 +10,39 @@ import { useRef } from '@wordpress/element';
  */
 import { STORE_NAME } from './constants';
 
-export const withOptionsHydration = ( data ) =>
-	createHigherOrderComponent(
-		( OriginalComponent ) => ( props ) => {
-			const dataRef = useRef( data );
+export const useOptionsHydration = ( data ) => {
```

Comment:
Added a hook for it as well, which I was going to use, but then didn't need to in the end.